File: LGPL/CommonSoftware/containerTests/corbaRefPersistenceTest/test/componentGetterImpl.cpp
#include "componentGetterImpl.h"

/* Cons + dest */
ComponentGetterImpl::ComponentGetterImpl(
		       const ACE_CString &name,
		       maci::ContainerServices * containerServices) :
    ACSComponentImpl(name, containerServices) { }
ComponentGetterImpl::~ComponentGetterImpl() {}

void ComponentGetterImpl::getOtherComponent() {
	getContainerServices()->getComponent<demo::ComponentGetter>("COMP_TO_GET");
	ACS_TRACE("Got component without problems :)");
}

#include <maciACSComponentDefines.h>
MACI_DLL_SUPPORT_FUNCTIONS(ComponentGetterImpl);

File: ACS/LGPL/CommonSoftware/containerTests/corbaRefPersistenceTest/test/demoImpl/ComponentGetterImpl.py
import demo__POA

from Acspy.Servants.ContainerServices  import ContainerServices
from Acspy.Servants.ComponentLifecycle import ComponentLifecycle
from Acspy.Servants.ACSComponent       import ACSComponent

class ComponentGetterImpl(demo__POA.ComponentGetter,
                          ACSComponent,
                          ContainerServices,
                          ComponentLifecycle):

	def __init__(self):
		ACSComponent.__init__(self)
		ComponentLifecycle.__init__(self)
		ContainerServices.__init__(self)

	def getOtherComponent(self):
		self.getComponent("COMP_TO_GET")
		self.getLogger().logInfo("Got component without problems :)");
